packages/trpc/server/routers/viewer/teams/getUserConnectedApps.handler.ts

```diff
@@ -0,0 +1,91 @@
+import { Prisma } from "@prisma/client";
+
+import { getAppFromSlug } from "@calcom/app-store/utils";
+import type { PrismaClient } from "@calcom/prisma";
+import type { AppCategories } from "@calcom/prisma/enums";
+import type { TrpcSessionUser } from "@calcom/trpc/server/trpc";
+
+import type { TGetUserConnectedAppsInputSchema } from "./getUserConnectedApps.schema";
+
+type GetUserConnectedAppsOptions = {
+  ctx: {
+    user: NonNullable<TrpcSessionUser>;
+    prisma: PrismaClient;
+  };
+  input: TGetUserConnectedAppsInputSchema;
+};
+
+const credentialSelect = Prisma.validator<Prisma.CredentialSelect>()({
+  userId: true,
+  app: {
+    select: {
+      slug: true,
+      categories: true,
+    },
+  },
+  destinationCalendars: {
+    select: {
+      externalId: true,
+    },
+  },
+});
+
+type Credential = Prisma.CredentialGetPayload<{ select: typeof credentialSelect }>;
+
+type Apps = {
+  name: string | null;
+  logo: string | null;
+  externalId: string | null;
+  app: { slug: string; categories: AppCategories[] } | null;
+};
+
+// This should improve performance saving already app data found.
+const appDataMap = new Map();
+
+export const getUserConnectedAppsHandler = async ({ ctx, input }: GetUserConnectedAppsOptions) => {
+  const { userIds } = input;
+
+  const credentialsPromises: Promise<Credential[]>[] = [];
+  const userConnectedAppsMap: Record<number, Apps[]> = {};
+
+  for (const userId of userIds) {
+    const cred = ctx.prisma.credential.findMany({
```

Comment:
We don't seem to be authenticating/autherizing user.

Reply:
Blocker: Verified, that I could access any user's installed apps list once I am logged in.

Reply:
Yeah i thought of this but we aren't exposing any confidential information or any access token. we only this

```
type Apps = {
  name: string | null;
  logo: string | null;
  externalId: string | null;
  app: { slug: string; categories: AppCategories[] } | null;
};

Reply:
IMHO, We still shouldn't allow it. Someone might start adding more stuff at that endpoint.

Unless that endpoint makes it clear that it is to be used by all logged in users, it is a no for such cases.

Also, for some people it could be important to not leak even this information

Reply:
Forgot about this. I'll fit it.